sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Analytics.Synapse.Spark.Models;
+
+namespace Azure.Analytics.Synapse.Spark
+{
+    /// <summary>
+    /// A long-running operation for:
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJobAsync(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJob(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJobAsync(int, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJob(int, bool?, CancellationToken)"/>
+    /// </summary>
+    public class SparkBatchOperation : Operation<SparkBatchJob>
+    {
+        private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(5);
+
+        private readonly ClientDiagnostics _diagnostics;
+
+        private readonly SparkBatchClient _client;
+        private readonly SparkBatchJob _value;
+        private Response<SparkBatchJob> _response;
+        private bool _completed;
+
+        private static List<string> SessionSubmissionFinalStates = new List<string>
+        {
+            "error",
+            "dead",
+            "success",
+            "killed"
+        };
+
+        internal SparkBatchOperation(SparkBatchClient client, ClientDiagnostics diagnostics, Response<SparkBatchJob> response)
+        {
+            _client = client;
+            _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
+            _response = response;
+            _completed = false;
+            _diagnostics = diagnostics;
+        }
+
+        /// <inheritdoc/>
+        public override string Id => _value.Id.ToString(CultureInfo.InvariantCulture);
+
+        /// <summary>
+        /// Gets the <see cref="SparkBatchJob"/>.
+        /// You should await <see cref="WaitForCompletionAsync(CancellationToken)"/> before attempting to use session in this pending state.
+        /// </summary>
+        /// <remarks>
+        /// Azure Synapse will return a <see cref="SparkBatchJob"/> immediately but may take time to the session to be ready.
+        /// </remarks>
+        public override SparkBatchJob Value => _value;
```

Comment:
This should throw under the conditions specified in the guidelines https://azure.github.io/azure-sdk/dotnet_implementation.html#dotnet-implement-operation

Reply:
Done.